explorer/src/components/ContentCard.tsx

```diff
@@ -0,0 +1,28 @@
+import { Card, CardHeader, CardContent, Typography } from '@mui/material';
+import React from 'react';
+
+export const ContentCard: React.FC<{
+  title: string;
+  subtitle?: string;
+  Icon?: React.ReactNode;
+  Action?: React.ReactNode;
+  errorMsg?: string;
+}> = ({ title, Icon, Action, subtitle, errorMsg, children }) => (
+  <Card sx={{ m: 3 }}>
+    <CardHeader
+      title={title}
+      avatar={Icon}
+      action={Action}
+      subheader={subtitle}
+    />
+    {children && <CardContent>{children}</CardContent>}
+    {errorMsg && (
+      <Typography
+        variant="body2"
+        sx={{ color: 'red', padding: (theme) => theme.spacing(2) }}
```

Comment:
```suggestion
        sx={{ color: 'danger', padding: 2 }}
```

`padding: 2` is the same as `padding: (theme) => theme.spacing(2)` just shorter

And best to use the semantic names for colours i.e. show the user something bad happened, rather than `goldenrod === warning`